serde/src/ser/mod.rs

```diff
@@ -202,53 +206,60 @@ pub trait Serializer {
     ///
     /// Callees of this method need to construct a `SeqVisitor`, which iterates through each item
     /// in the sequence.
-    fn serialize_seq<V>(&mut self, visitor: V) -> Result<(), Self::Error>
-        where V: SeqVisitor;
+    fn serialize_seq(&mut self, len: Option<usize>) -> Result<(), Self::Error>;
 
     /// Serializes a sequence element.
     fn serialize_seq_elt<T>(&mut self, value: T) -> Result<(), Self::Error>
         where T: Serialize;
 
+    /// Finish serializing a sequence.
+    fn serialize_seq_end(&mut self, len: Option<usize>) -> Result<(), Self::Error>;
+
     /// Serializes a tuple.
     ///
     /// By default this serializes a tuple as a sequence.
     #[inline]
-    fn serialize_tuple<V>(&mut self, visitor: V) -> Result<(), Self::Error>
-        where V: SeqVisitor,
+    fn serialize_tuple(&mut self, len: usize) -> Result<(), Self::Error>
     {
-        self.serialize_seq(visitor)
+        self.serialize_seq(Some(len))
     }
 
     /// Serializes a tuple element.
     ///
     /// By default, tuples are serialized as a sequence.
     #[inline]
     fn serialize_tuple_elt<T>(&mut self, value: T) -> Result<(), Self::Error>
-        where T: Serialize
-    {
+        where T: Serialize {
```

Comment:
I liked this before.

Looks like you meant to move the brace of the previous function, not this one.